tests/statistic_reporter_test.py

```diff
@@ -99,10 +101,32 @@ def poll(self, _):
     assert update_buffer_err_counter.value == 1
 
 
+def test_producer_increments_delivery_error_counter_on_delivery_error():
+    class FakeProducer:
+        def produce(self, topic, payload, key, on_delivery, timestamp):
+            error = KafkaError(KafkaError.INVALID_CONFIG)
+            on_delivery(error, "some error message")
+
+        def flush(self, _):
+            pass
+
+        def poll(self, _):
+            pass
+
+    update_delivery_err_counter: Counter = Counter()
+    kafka_producer = KafkaProducer(
+        FakeProducer(), update_delivery_err_counter=update_delivery_err_counter
+    )
+
+    kafka_producer.produce("IRRELEVANT_TOPIC", b"IRRELEVANT_PAYLOAD", 0, key="PV_NAME")
+    kafka_producer.close()
+    assert update_delivery_err_counter.value == 1
+
+
 def test_statistic_reporter_sends_data_loss_errors():
     update_buffer_err_counter: Counter = Counter()
     statistics_reporter = StatisticsReporter(
-        "localhost", {}, Counter(), update_buffer_err_counter, logger
+        "localhost", {}, Counter(), update_buffer_err_counter, Counter(), logger
```

Comment:
ok, I see the pattern repeating, this probably means that it is something I don't understand :-)